addons/project/controllers/portal.py

```diff
@@ -321,11 +321,11 @@ def _prepare_tasks_values(self, page, date_begin, date_end, sortby, search, sear
         searchbar_groupby = self._task_get_searchbar_groupby()
 
         Task = request.env['project.task']
-        if su:
-            Task = Task.sudo()
-
         if not domain:
             domain = []
+        if not su:
```

Comment:
We could check if the user can read the tasks because if one day this method is used in another route and it doesn't check if the current user is a public one without any access_token.
```suggestion
        if not su and Task.check_access_rights('read'):
```